src/config.js

```diff
@@ -212,6 +225,59 @@ function isValidJsonSchema(schema, spec) {
   return true;
 }
 
+function validateStackGroups(stackGroups, stacks) {
+  _.flow(
+    (stackGroups) => {
+      if (_.isUndefined(stackGroups)) {
+        return [];
+      }
+      if (!_.isArray(stackGroups)) {
+        throw new Error('stackGroups argument must be an array of objects or undefined');
+      }
+      _.forEach(stackGroups, (stackGroup) => {
+        if (!_.isString(stackGroup.name)) {
+          throw new Error('stackGroup.name must be a string');
+        }
+        if (!_.isArray(stackGroup.stacks)) {
+          throw new Error('stackGroup.stacks must be an array of strings');
+        }
+      });
+      const duplicateGroups = _.chain(stackGroups)
+        .groupBy('name')
+        .filter((value, key) => value.length > 1)
+        .map((x) => x[0].name)
+        .value();
+      if(duplicateGroups.length > 0) {
+        throw new Error(`the following stackGroup(s) can only be defined once: ${duplicateGroups.join(', ')}`);
+      }
+
+      return stackGroups;
+    },
+    _.partialRight(_.flatMap, (stackGroup) => _.map(stackGroup.stacks, (stack) => [stack, stackGroup.name])),
+    _.flow(
+      _.partialRight(_.groupBy, (x) => x[0]),
+      (group) => {
+        const invalidStacks = _.flow(
+          _.keys,
+          _.partialRight(_.filter, (stackName) => !_.find(stacks, (x) => x.name == stackName)),
+        )(group);
+        if(invalidStacks.length > 0) {
+          throw new Error(`the following stack(s) need to be defined in the stacks file: ${invalidStacks.join(', ')}`);
+        }
+
+        return group;
+      },
+      _.partialRight(_.pickBy, (v, k) => v.length > 1),
+      _.keys,
+      (invalidStacks) => {
+        if(invalidStacks.length > 0) {
+          throw new Error(`the following stack(s) cannot appear in more than one group: ${invalidStacks.join(', ')}`);
+        }
+      },
+    ),
+  )(stackGroups);
+}
```

Comment:
This is good/thorough. If we continue to invest in nodecf (vs a CDK-based approach), it'd probably be worth lifting each of the enhancers/operators of the chain into independent functions and name in such a way that the validation process is a bit more self-descriptive.

It's really not that bad, but you do need to read the code to infer intent.